src/components/Reown/modals/SendNanoContractTxModal.js

```diff
@@ -0,0 +1,187 @@
+                        )}
+                      </td>
+                    </tr>
+                  ))}
+                </tbody>
+              </table>
+            </div>
+          </div>
+        </div>
+      </>
+    );
+  };
+
+  return (
+    <>
+      <div className="modal-header">
+        <h5 className="modal-title">{t`NEW NANO CONTRACT TRANSACTION`}</h5>
+        <button type="button" className="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+      <div className="modal-body">
+        <div className="d-flex align-items-center mb-3">
+          {data.dapp.icon && <img src={data.dapp.icon} alt="dApp icon" className="mr-3" style={{ width: 48, height: 48 }} />}
+          <div>
+            <h6 className="mb-1">{data.dapp.proposer}</h6>
+            <small className="text-muted">{data.dapp.url}</small>
+          </div>
+        </div>
+
+        {/* Blueprint Information Card */}
+        <div className="card mb-4">
+          <div className="card-body">
+            {data.data.ncId && (
+              <div className="mb-3">
+                <strong>{t`Nano Contract ID`}</strong>
+                <div className="text-monospace">
+                  {helpers.truncateText(data.data.ncId, 8, 4)}
+                  <button 
+                    className="btn btn-link btn-sm p-0 ml-2" 
+                    onClick={() => navigator.clipboard.writeText(data.data.ncId)}
+                  >
+                    <i className="fa fa-copy"></i>
+                  </button>
+                </div>
+              </div>
+            )}
+
+            <div className="mb-3">
+              <strong>{t`Blueprint ID`}</strong>
+              <div className="text-monospace">
+                {helpers.truncateText(data.data.blueprintId, 8, 4)}
+                <button 
+                  className="btn btn-link btn-sm p-0 ml-2" 
+                  onClick={() => navigator.clipboard.writeText(data.data.blueprintId)}
+                >
+                  <i className="fa fa-copy"></i>
+                </button>
+              </div>
+            </div>
+
+            <div className="mb-3">
+              <strong>{t`Blueprint Name`}</strong>
+              <div>{blueprintInfo?.name || '-'}</div>
+            </div>
+
+            <div className="mb-3">
+              <strong>{t`Blueprint Method`}</strong>
+              <div>{data.data.method}</div>
+            </div>
+
+            <div className="mb-3">
+              <strong>{t`Caller`}</strong>
+              <div className="text-monospace">
+                {helpers.truncateText(firstAddress, 8, 4)}
+                <button 
+                  className="btn btn-link btn-sm p-0 ml-2" 
+                  onClick={() => navigator.clipboard.writeText(firstAddress)}
+                >
+                  <i className="fa fa-copy"></i>
+                </button>
+              </div>
+            </div>
```

Comment:
Why is it hardcoded as `firstAddress`? The user might change this address if the nano is registered in the wallet.